### PR TITLE
DC-3095: wire create conversation API to core conversation model

### PR DESCRIPTION
**Pull request by rwalpole**

### CODE REVIEW

File: public/schema.json [hunk 1]
```diff
@@ -46,54 +46,56 @@
...
         "properties" : {
           "sender" : {
-            "$ref" : "#/components/schemas/uk.gov.hmrc.securemessage.controllers.model.cdcm.Sender"
+            "$ref" : "#/components/schemas/uk.gov.hmrc.securemessage.controllers.models.generic.Sender"
           },
           "recipients" : {
             "type" : "array",
             "items" : {
-              "$ref" : "#/components/schemas/uk.gov.hmrc.securemessage.controllers.model.cdcm.Recipient"
+              "$ref" : "#/components/schemas/uk.gov.hmrc.securemessage.controllers.models.generic.Recipient"
```
Comment (elabeca) [inline public/schema.json:91]: from model.cdcm to model.generic, is that intended? I know Matt had that in reverse

File: test/uk/gov/hmrc/securemessage/controllers/models/generic/ConversationRequestSpec.scala [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+  "A full conversation API request model" should {
+    "be converted correctly to a core conversation model" in {
+      val formatter = ISODateTimeFormat.dateTime()
+      val dateInString = "2020-11-10T15:00:01.000Z"
+      val dateTime = DateTime.parse(dateInString, formatter)
+      val fullConversationRequestJson: JsValue = Resources.readJson("model/api/create-conversation-full.json")
+      fullConversationRequestJson.validate[ConversationRequest] match {
+        case s: JsSuccess[ConversationRequest] =>
+          val conversationRequest: ConversationRequest = s.get
```
Comment (elabeca) [inline test/uk/gov/hmrc/securemessage/controllers/models/generic/ConversationRequestSpec.scala:36]: WartRemover should fail on this call, surprised it passed

**Review by elabeca — DISMISSED** [review]

File: public/schema.json (hunk 1 above)
Comment (rwalpole) [reply public/schema.json:91]: Yes it is intended. Oleg specifically requested that we change this as it is no longer CDCM specific following discussions earlier today.

File: test/uk/gov/hmrc/securemessage/controllers/models/generic/ConversationRequestSpec.scala (hunk 2 above)
Comment (rwalpole) [reply test/uk/gov/hmrc/securemessage/controllers/models/generic/ConversationRequestSpec.scala]: @elabeca I think Wartremover only checks `Options`, `Either` and `Try` for `.get` whereas this is a `JsSuccess`. I fixed it anyway.

**Review by elabeca — APPROVED** [review]

**Pull request merged**